src/main/java/seedu/address/logic/commands/UndoCommand.java

```diff
@@ -12,22 +12,18 @@
 
     public static final String COMMAND_WORD = "undo";
     public static final String MESSAGE_USAGE_SUCCESS = "Undo success!";
-    public static final String MESSAGE_USAGE_FAILURE = "No more commands to undo!";
+    public static final String MESSAGE_USAGE_FAILURE = "Unable to perform undo!";
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireAllNonNull(model, undoRedoStack);
 
         if (!undoRedoStack.canUndo()) {
             throw new CommandException(MESSAGE_USAGE_FAILURE);
+        } else {
+            undoRedoStack.popUndo().undo(model);
+            return new CommandResult(MESSAGE_USAGE_SUCCESS);
         }
```

Comment:
Removing the check for SwitchCommand breaks undo, redo for that command.

Reply:
Yep, can confirm this causes the bug. To reproduce:
1. Switch
2. Undo
3. States of `archivedAddressBook.json` and `addressbook.json` gets messed up.